dev-client/src/screens/UserSettingsScreen/components/menu/SignOutItem.tsx

```diff
@@ -16,28 +16,38 @@
  */
 
 import {useTranslation} from 'react-i18next';
+import {PressableProps} from 'react-native';
 
 import {MenuItem} from 'terraso-mobile-client/components/menus/MenuItem';
 import {SignOutModal} from 'terraso-mobile-client/components/modals/SignOutModal';
 import {useIsOffline} from 'terraso-mobile-client/hooks/connectivityHooks';
 
-export function SignOutItem() {
-  const {t} = useTranslation();
+type Props = {
+  disabled?: boolean;
+  onPress?: PressableProps['onPress'];
+};
 
-  const isDisabled = useIsOffline();
+const SignOutMenuItem = ({disabled, onPress}: Props) => {
+  const {t} = useTranslation();
 
   return (
-    <SignOutModal
-      trigger={onOpen => (
-        <MenuItem
-          variant="default"
-          uppercase
-          icon="logout"
-          label={t('settings.sign_out')}
-          disabled={isDisabled}
-          onPress={onOpen}
-        />
-      )}
+    <MenuItem
+      variant="default"
+      uppercase
+      icon="logout"
+      label={t('settings.sign_out')}
+      disabled={disabled}
+      onPress={onPress}
     />
   );
-}
+};
+
+export const SignOutItem = () => {
+  const isDisabled = useIsOffline();
+
+  return isDisabled ? (
+    <SignOutMenuItem disabled={isDisabled} />
+  ) : (
+    <SignOutModal trigger={onOpen => <SignOutMenuItem onPress={onOpen} />} />
```

Comment:
Note: Hmm the modal/trigger hierarchy here makes this read particularly awkward. We may want to rethink our pattern here if we're going to have to end up doing this in a lot of different places.

Reply:
I couldn't find a way to disable the trigger itself. I tried disabling onPress, but that didn't work.

Reply:
@tm-ruxandra any suggestions?

Reply:
@paulschreiber There's probably a better solution here but it requires re-working how our "trigger" system works. This is fine for now.